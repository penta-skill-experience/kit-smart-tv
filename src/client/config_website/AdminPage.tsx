import * as React from "react";
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import {Grid} from "@mui/material";
import InputLabel from "@mui/material/InputLabel";
import Input from "@mui/material/Input";
import InputAdornment from "@mui/material/InputAdornment";
import IconButton from "@mui/material/IconButton";
import VisibilityOff from "@mui/icons-material/VisibilityOff";
import Visibility from "@mui/icons-material/Visibility";
import FormControl from "@mui/material/FormControl";
import Snackbar from "@mui/material/Snackbar";

export const AdminPage = ({oldPassword, newPassword, handleOldPassword, handleNewPassword, handlePasswordChange, children} ) => {
    const [showOldPassword, setShowOldPassword] = React.useState(false);
    const [showNewPassword, setShowNewPassword] = React.useState(false);

    const handleShowOldPassword  = () => {
        setShowOldPassword(!showOldPassword)
    }

    const handleShowNewPassword  = () => {
        setShowNewPassword(!showNewPassword)
    }

    const [successfulBar, setSuccessfulBar] = React.useState(false);
    const [errorBar, setErrorBar] = React.useState(false);

    const handleClose = (event: React.SyntheticEvent | Event, reason?: string) => {
        if (reason === 'clickaway') {
            return;
        }
        setSuccessfulBar(false);
        setErrorBar(false);
    }

    return (
        <Box
        sx={{
            width: 300,
            height: 300,
            }
        }
        >
            <div>
                <Grid container spacing={2} direction="row" justifyContent="center" alignItems="center">
                    <Grid item xs={12}>
                        <FormControl sx={{ m: 1, width: '25ch' }} variant="standard">
                            <InputLabel>Old Password</InputLabel>
                            <Input
                                type={showOldPassword ? 'text' : 'password'}
                                value={oldPassword}
                                onChange={handleOldPassword}
                                endAdornment={
                                    <InputAdornment position="end">
                                        <IconButton
                                            onClick={handleShowOldPassword}
                                        >
                                            {showOldPassword ? <VisibilityOff /> : <Visibility />}
                                        </IconButton>
                                    </InputAdornment>
                                }
                            />
                        </FormControl>
                    </Grid>
                    <Grid item xs={12}>
                        <FormControl sx={{ m: 1, width: '25ch' }} variant="standard">
                            <InputLabel>New Password</InputLabel>
                                <Input
                                type={showNewPassword ? 'text' : 'password'}
                                value={newPassword}
                                onChange={handleNewPassword}
                                endAdornment={
                                    <InputAdornment position="end">
                                        <IconButton
                                        onClick={handleShowNewPassword}
                                        >
                                        {showNewPassword ? <VisibilityOff /> : <Visibility />}
                                        </IconButton>
                                    </InputAdornment>
                                }
                                />
                        </FormControl>
                    </Grid>
                    <Grid item xs={12}>
                        <Button
                            onClick={() => {
                                if (handlePasswordChange()) {
                                    setSuccessfulBar(true);
                                } else{
                                    setErrorBar(true);
                                }
                            }}
                            variant="outlined"
                        >
                            Save Password
                        </Button>
                        <Snackbar
                            open={successfulBar}
                            autoHideDuration={6000}
                            onClose={handleClose}
                            message={'Password Saved'}
                        />
                        <Snackbar
                            open={errorBar}
                            autoHideDuration={6000}
                            onClose={handleClose}
                            message={'New Password must at least be 1 character long'}
                        />
                    </Grid>
                    <Grid item xs={12}>
                    </Grid>
                </Grid>
            </div>
        </Box>
    );
}