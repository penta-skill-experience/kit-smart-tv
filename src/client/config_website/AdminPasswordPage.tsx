import * as React from "react";
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';

export const AdminPasswordPage = ({oldPassword, newPassword, handleOldPassword, handleNewPassword, handlePasswordChange, children} ) => {

    return (
        <Box
        sx={{
            width: 300,
            height: 300,
            }
        }
        >
            <div>
                <TextField
                    value={oldPassword}
                    label="Old Password"
                    onChange={handleOldPassword}
                />
                <TextField
                    value={newPassword}
                    label="New Password"
                    onChange={handleNewPassword}
                />
                <Button
                    onClick={handlePasswordChange}
                    variant="outlined"
                >
                    Save Password</Button>
            </div>
        </Box>
    );

}