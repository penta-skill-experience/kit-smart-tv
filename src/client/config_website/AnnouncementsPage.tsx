import * as React from "react";
import {Grid} from "@mui/material";
import TextField from '@mui/material/TextField';
import Button from "@mui/material/Button";
import DeleteIcon from "@mui/icons-material/Delete";

export const AnnouncementsPage = () => {
    const initialMailList = [];
    const [mailList, setMailList] = React.useState(initialMailList);
    const [verUser, setVerUser] = React.useState({
        mail:'',
        name:'',
    });


    const handleMailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, mail:event.target.value});
    };

    const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, name:event.target.value});
    };

    const handleAddMail = () => {
        if (verUser.name !== '' && verUser.mail !== '') {
            const newUser = {
                mail:verUser.mail,
                name:verUser.name,
            }
            setMailList(mailList.concat(newUser));
        }
        alert('Username and email have to be filled out')
    };

    const handleDeleteUser = (mail) => {
        setMailList(mailList.filter(item => item.mail !== mail));
    }


    return(
        <div>
            <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="center">
                <Grid item xs={12}>
                    <h1>Add e-mail as verified: </h1>
                </Grid>
                <Grid item>
                    <TextField
                        label="E-Mail"
                        value={verUser.mail}
                        onChange={handleMailChange}
                    />
                </Grid>
                <Grid item>
                    <TextField
                        label="Name"
                        value={verUser.name}
                        onChange={handleNameChange}
                    />
                </Grid>
                <Grid item>
                    <Button variant="outlined" onClick={handleAddMail}>
                        Add
                    </Button>
                </Grid>
                <Grid item xs={12}>
                    <h1>Currently verified accounts:</h1>
                </Grid>
                <Grid item xs={4}>
                    Email
                </Grid>
                <Grid item xs={2}>
                    Name
                </Grid>
                <Grid item xs={6}></Grid>
                <Grid item xs={12}>
                    <ul>
                        {mailList.map(item => (
                            <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="center">
                                <Grid item xs={4}>
                                    {item.mail}
                                </Grid>
                                <Grid item xs={1}>
                                    {item.name}
                                </Grid>
                                <Grid item xs={1}>
                                    <Button onClick={() => {handleDeleteUser(item.mail)}}>
                                        <DeleteIcon/>
                                    </Button>
                                </Grid>
                            </Grid>
                        ))}
                    </ul>
                </Grid>
            </Grid>
        </div>
    );
}