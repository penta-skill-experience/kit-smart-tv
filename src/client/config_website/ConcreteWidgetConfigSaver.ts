import {WidgetConfigSaver} from "../widget/WidgetConfigSaver";

export class ConcreteWidgetConfigSaver implements WidgetConfigSaver {

    save(rawConfig: Object): void {

        // todo write to persistence
        console.log("save widget config:");
        console.log(rawConfig);
    }
}