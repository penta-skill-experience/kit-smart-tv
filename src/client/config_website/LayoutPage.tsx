import * as React from "react";
import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import Select from '@mui/material/Select';
import {Button, Grid} from "@mui/material";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import LockIcon from '@mui/icons-material/Lock';
import {WidgetListElement} from "./WidgetListElement";
import {WidgetLoader} from "../widget/WidgetLoader";
import Snackbar from "@mui/material/Snackbar";


const widgetLoader = new WidgetLoader();
const widgetList = widgetLoader.getWidgetIds();


export const LayoutPage = ({list, widgetListElement, handleWidgetSelection, handleAddWidget, handleDeleteWidget, handlePosition, handleRawConfigSave, handleLayoutChange, children}) => {

    const [open, setOpen] = React.useState(false);

    const handleClick = () => {
        setOpen(true);
    };

    const handleClose = (event: React.SyntheticEvent | Event, reason?: string) => {
        if (reason === 'clickaway') {
            return;
        }
        setOpen(false);
    }

    return(
        <div>
            <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="flex-start">
                <Grid item container spacing={2} direction="row" xs={12}>
                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                            }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <LockIcon color='error'/>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>

                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                                }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>1.top center</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>

                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                            }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>2.top right</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>

                </Grid>

                <Grid item container spacing={2} direction="row">
                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                            }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>3.bottom left</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>

                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                            }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>4.bottom center</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>

                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                            }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>5.bottom right</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>
                </Grid>

                <Grid item xs={12}>
                    <ul>
                        {list.map(item => (
                            <WidgetListElement
                                item={item}
                                handlePosition={handlePosition}
                                handleDeleteWidget={handleDeleteWidget}
                                handleRawConfigSave={handleRawConfigSave}
                            >
                            </WidgetListElement>
                        ))}
                    </ul>
                </Grid>
                <Grid item xs={12}>
                    <FormControl sx={{minWidth: 120}}>
                        <Select
                            value={widgetListElement.widgetNameText}
                            onChange={handleWidgetSelection}
                        >
                            {widgetList.map(item => (
                                <MenuItem value={item}>
                                    {widgetLoader.getWidget(item).getTitle()}
                                </MenuItem>
                            ))}
                        </Select>
                    </FormControl>
                </Grid>
                <Grid item xs={12}>
                    <Button onClick={handleAddWidget} variant="outlined">Add Widget</Button>
                </Grid>
                <Grid item>
                    <Button variant="contained" onClick={() => {
                        handleLayoutChange();
                        handleClick();
                    }}>
                        Save
                    </Button>
                    <Snackbar
                        open={open}
                        autoHideDuration={6000}
                        onClose={handleClose}
                        message={'Changes Saved'}
                    />
                </Grid>
            </Grid>
        </div>
    );
};