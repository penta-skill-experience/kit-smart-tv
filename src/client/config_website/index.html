<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Admin Interface</title>
    <meta content="width=device-width, initial-scale=1" name="viewport">
    <link
            href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
            rel="stylesheet"
    />
    <style>
        #root {
            height: 100vh;
            width: 100vw;
        }
    </style>
</head>
<body>
<div id="root">
    <!-- react output will go inside here -->
</div>
</body>
</html>
