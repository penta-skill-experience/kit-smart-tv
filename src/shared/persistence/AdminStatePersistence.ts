import * as https from "https";
import config from "./persistence.config.json";

export class AdminStatePersistence {
    get refresh_token(): string {
        return this._refresh_token;
    }
    get access_token(): string {
        return this._access_token;
    }
    private _access_token: string;
    private _refresh_token: string;
    constructor() {
    }

    async login(password: string) : Promise<boolean> {
        const response = await fetch(`${config.DOMAIN}/api/sessions`, {
            method: 'POST',
            body: JSON.stringify(password), // string or object
            headers: {
                'Content-Type': 'application/json'
            }
        });
        try{
            const myJson = await response.json(); //extract JSON from the http response
            if(myJson){
                this._access_token = myJson.accessToken;
                this._refresh_token = myJson.refreshToken;
                return true;
            }
        }
        catch(e: any)
        {
            return false;
        }
        return false;
    }

    async logout(access_token: string, refresh_token: string): Promise<boolean> {
        const response = await fetch(`${config.DOMAIN}/api/sessions`, {
            method: 'DELETE',
            body: null, // string or object
            headers: {
                'Content-Type': 'application/json',
                'x-refresh': refresh_token,
                'Authorization': 'Bearer ' + access_token,
    }
        });
        try{
            const myJson = await response.json(); //extract JSON from the http response
            if(myJson){
                this._access_token = myJson.accessToken;
                this._refresh_token = myJson.refreshToken;
                return true;
            }
        }
        catch(e: any)
        {
            return false;
        }
        return false;
    }

    async getAdminLoginState(access_token: string, refresh_token: string): Promise<boolean> {
        const response = await fetch(`${config.DOMAIN}/api/sessions`, {
            method: 'GET',
            body: null,
            headers: {
                'Content-Type': 'application/json',
                'x-refresh': refresh_token,
                'Authorization': 'Bearer ' + access_token,
            }
        });
        try{
            const myJson = await response.json(); //extract JSON from the http response
            if(new Date(myJson.valid_until).valueOf() > Date.now()){
                return true;
            }
        }
        catch(e: any)
        {
            return false;
        }
        return false;
    }

    async setPassword(oldpw: string, newpw: string, access_token: string, refresh_token: string): Promise<boolean> {
        const response = await fetch(`${config.DOMAIN}/admin/update-password`, {
            method: 'PUT',
            body: JSON.stringify({password: oldpw, new_password: newpw}),
            headers: {
                'Content-Type': 'application/json',
                'x-refresh': refresh_token,
                'Authorization': 'Bearer ' + access_token,
            }
        });
        try { //extract JSON from the http response
            if ( "password updated" === JSON.stringify(response.body)) {
                return true;
            }
        } catch (e: any) {
            return false;
        }
        return false;
    }
}