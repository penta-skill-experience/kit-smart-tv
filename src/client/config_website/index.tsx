import "./index.css";
import * as ReactDOM from "react-dom";
import * as React from "react";
import {WidgetLoader} from "../widget/WidgetLoader";
import {ConcreteWidgetConfigSaver} from "./ConcreteWidgetConfigSaver";

ReactDOM.render(
    // this is just an example to demonstrate how widgets are instantiated
    // todo: remove this.
    <div>
        <h1>Hello Admin!</h1>
        <div>
            {new WidgetLoader().getWidget("rss-feed").createConfigComponent({url: "www.example.com"}, new ConcreteWidgetConfigSaver())}
        </div>
    </div>,
    document.getElementById("root")
);