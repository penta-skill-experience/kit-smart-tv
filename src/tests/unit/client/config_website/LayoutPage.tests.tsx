import {shallow, configure} from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {LayoutPage} from "../../../../client/config_website/LayoutPage";
import * as React from "react";
import toJson from "enzyme-to-json";
import {AdminStatePersistence} from "../../../../shared/persistence/AdminStatePersistence";

configure({adapter: new Adapter()});

describe("Layout page tests", () => {
    let wrapper;
    const getAdminLoginStateMock = jest.spyOn(AdminStatePersistence.prototype, "getAdminLoginState");
    const widgetListElement = {
        id:0,
        position:'',
        widgetNameText:'',
        widget: null,
        widgetData:null,
    };

    const list = [widgetListElement];

    const method = () => {
    };



    beforeEach(() => {
       wrapper = shallow(
           <LayoutPage
               list={list}
               widgetListElement={widgetListElement}
               handleWidgetSelection={method}
               handleAddWidget={method}
               handleDeleteWidget={method}
               handlePosition={method}
               handleRawConfigSave={method}
               handleLayoutChange={method}
           >

           </LayoutPage>
       );
    });

    test("snapshot test for layout page", async () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test("save button test", () => {
        getAdminLoginStateMock.mockImplementation(() => {
            return new Promise<void>(resolve => {
                resolve();
            });
        })
        wrapper.find("ForwardRef(Button)").at(1).simulate("click");
        expect(getAdminLoginStateMock).toHaveBeenCalled();
    });

    test("handleClose function test", () => {
        wrapper.find("ForwardRef(Snackbar)").at(0).simulate("close");
        expect(wrapper.find("ForwardRef(Snackbar)").at(0).prop("open")).toEqual(false);
    });

    test("handleClose function if case test", () => {
        wrapper.find("ForwardRef(Snackbar)").at(0).simulate("close", { target: { event: '', reason: 'clickaway'} });
        expect(wrapper.find("ForwardRef(Snackbar)").at(0).prop("open")).toEqual(false);
    });

});