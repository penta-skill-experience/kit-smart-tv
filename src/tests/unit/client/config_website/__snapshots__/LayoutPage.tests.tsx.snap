// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout page tests snapshot test for layout page 1`] = `
<div>
  <ForwardRef(Grid)
    alignItems="flex-start"
    container={true}
    direction="row"
    justifyContent="flex-start"
    spacing={2}
  >
    <ForwardRef(Grid)
      container={true}
      direction="row"
      item={true}
      spacing={2}
      xs={12}
    >
      <ForwardRef(Grid)
        item={true}
      >
        <ForwardRef(Box)
          sx={
            Object {
              "backgroundColor": "text.primary",
              "height": 150,
              "width": 150,
            }
          }
        >
          <ForwardRef(Grid)
            alignItems="center"
            container={true}
            direction="row"
            justifyContent="space-between"
          >
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              alignItems="center"
              container={true}
              item={true}
              justifyContent="center"
            >
              <ForwardRef(Grid)
                item={true}
              >
                <Memo(ForwardRef(LockIcon))
                  color="error"
                />
              </ForwardRef(Grid)>
            </ForwardRef(Grid)>
          </ForwardRef(Grid)>
        </ForwardRef(Box)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
      >
        <ForwardRef(Box)
          sx={
            Object {
              "backgroundColor": "text.primary",
              "height": 150,
              "width": 150,
            }
          }
        >
          <ForwardRef(Grid)
            alignItems="center"
            container={true}
            direction="row"
            justifyContent="space-between"
          >
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              alignItems="center"
              container={true}
              item={true}
              justifyContent="center"
            >
              <ForwardRef(Grid)
                item={true}
              >
                <ForwardRef(Typography)
                  color="white"
                  variant="h6"
                >
                  1.top center
                </ForwardRef(Typography)>
              </ForwardRef(Grid)>
            </ForwardRef(Grid)>
          </ForwardRef(Grid)>
        </ForwardRef(Box)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
      >
        <ForwardRef(Box)
          sx={
            Object {
              "backgroundColor": "text.primary",
              "height": 150,
              "width": 150,
            }
          }
        >
          <ForwardRef(Grid)
            alignItems="center"
            container={true}
            direction="row"
            justifyContent="space-between"
          >
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              alignItems="center"
              container={true}
              item={true}
              justifyContent="center"
            >
              <ForwardRef(Grid)
                item={true}
              >
                <ForwardRef(Typography)
                  color="white"
                  variant="h6"
                >
                  2.top right
                </ForwardRef(Typography)>
              </ForwardRef(Grid)>
            </ForwardRef(Grid)>
          </ForwardRef(Grid)>
        </ForwardRef(Box)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      container={true}
      direction="row"
      item={true}
      spacing={2}
    >
      <ForwardRef(Grid)
        item={true}
      >
        <ForwardRef(Box)
          sx={
            Object {
              "backgroundColor": "text.primary",
              "height": 150,
              "width": 150,
            }
          }
        >
          <ForwardRef(Grid)
            alignItems="center"
            container={true}
            direction="row"
            justifyContent="space-between"
          >
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              alignItems="center"
              container={true}
              item={true}
              justifyContent="center"
            >
              <ForwardRef(Grid)
                item={true}
              >
                <ForwardRef(Typography)
                  color="white"
                  variant="h6"
                >
                  3.bottom left
                </ForwardRef(Typography)>
              </ForwardRef(Grid)>
            </ForwardRef(Grid)>
          </ForwardRef(Grid)>
        </ForwardRef(Box)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
      >
        <ForwardRef(Box)
          sx={
            Object {
              "backgroundColor": "text.primary",
              "height": 150,
              "width": 150,
            }
          }
        >
          <ForwardRef(Grid)
            alignItems="center"
            container={true}
            direction="row"
            justifyContent="space-between"
          >
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              alignItems="center"
              container={true}
              item={true}
              justifyContent="center"
            >
              <ForwardRef(Grid)
                item={true}
              >
                <ForwardRef(Typography)
                  color="white"
                  variant="h6"
                >
                  4.bottom center
                </ForwardRef(Typography)>
              </ForwardRef(Grid)>
            </ForwardRef(Grid)>
          </ForwardRef(Grid)>
        </ForwardRef(Box)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
      >
        <ForwardRef(Box)
          sx={
            Object {
              "backgroundColor": "text.primary",
              "height": 150,
              "width": 150,
            }
          }
        >
          <ForwardRef(Grid)
            alignItems="center"
            container={true}
            direction="row"
            justifyContent="space-between"
          >
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              item={true}
              xs={12}
            >
              ''
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              alignItems="center"
              container={true}
              item={true}
              justifyContent="center"
            >
              <ForwardRef(Grid)
                item={true}
              >
                <ForwardRef(Typography)
                  color="white"
                  variant="h6"
                >
                  5.bottom right
                </ForwardRef(Typography)>
              </ForwardRef(Grid)>
            </ForwardRef(Grid)>
          </ForwardRef(Grid)>
        </ForwardRef(Box)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ul>
        <WidgetListElement
          handleDeleteWidget={[Function]}
          handlePosition={[Function]}
          handleRawConfigSave={[Function]}
          item={
            Object {
              "id": 0,
              "position": "",
              "widget": null,
              "widgetData": null,
              "widgetNameText": "",
            }
          }
        />
      </ul>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(FormControl)
        sx={
          Object {
            "minWidth": 150,
          }
        }
      >
        <ForwardRef(InputLabel)>
          Select Widget
        </ForwardRef(InputLabel)>
        <ForwardRef(Select)
          key="0"
          label="Select Widget"
          onChange={[Function]}
          value=""
        >
          <ForwardRef(MenuItem)
            value="rss-feed"
          >
            RSS Feed
          </ForwardRef(MenuItem)>
          <ForwardRef(MenuItem)
            value="tram-schedule"
          >
            Tram Schedule
          </ForwardRef(MenuItem)>
          <ForwardRef(MenuItem)
            value="cafeteria-menu"
          >
            Cafeteria
          </ForwardRef(MenuItem)>
          <ForwardRef(MenuItem)
            value="announcement"
          >
            Announcements
          </ForwardRef(MenuItem)>
          <ForwardRef(MenuItem)
            value="cafeteria-opening"
          >
            Cafeteria Status
          </ForwardRef(MenuItem)>
        </ForwardRef(Select)>
      </ForwardRef(FormControl)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(Button)
        onClick={[Function]}
        variant="outlined"
      >
        Add Widget
      </ForwardRef(Button)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
    >
      <ForwardRef(Button)
        onClick={[Function]}
        variant="contained"
      >
        Save
      </ForwardRef(Button)>
      <ForwardRef(Snackbar)
        autoHideDuration={6000}
        message="Changes Saved"
        onClose={[Function]}
        open={false}
      />
      <ForwardRef(Snackbar)
        autoHideDuration={6000}
        message="Session expired"
        onClose={[Function]}
        open={false}
      />
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</div>
`;
